src/storage/src/storage_state/async_storage_worker.rs

```diff
@@ -247,81 +249,51 @@ impl<T: Timestamp + Lattice + Codec64 + Display> AsyncStorageWorker<T> {
                                 .open(persist_location.clone())
                                 .await
                                 .expect("error creating persist client");
+                            let diagnostics = Diagnostics {
+                                shard_name: id.to_string(),
+                                handle_purpose: format!("resumption data {}", id),
+                            };
 
                             let mut write_handle = client
                                 .open_writer::<SourceData, (), T, Diff>(
                                     *data_shard,
                                     Arc::new(relation_desc.clone()),
                                     Arc::new(UnitSchema),
-                                    Diagnostics {
-                                        shard_name: id.to_string(),
-                                        handle_purpose: format!("resumption data {}", id),
-                                    },
+                                    diagnostics.clone(),
                                 )
                                 .await
                                 .unwrap();
                             let upper = write_handle.fetch_recent_upper().await;
+
+                            let read_handle = client
+                                .open_critical_since::<SourceData, (), T, Diff, PersistEpoch>(
+                                    *data_shard,
+                                    PersistClient::CONTROLLER_CRITICAL_SINCE,
```

Comment:
Can't we use `open_leased_reader` with `use_critical_since` configured using the dyncfg?

I'm afraid using the specific critical since ID can break in subtle ways in the future: for example with the two-critical-handles-one-for-upgrades idea we recently discussed for 0dt, or in a future where clusters have their own critical since handle (id).

Reply:
Updated!

Reply:
merci! 🙇‍♂️